### PR TITLE
Upgrade test application to Sylius 1.12

### PR DESCRIPTION
**Pull request by JoppeDC**
- Upgrades test application to Sylius 1.12
- Upgrade Github Actions to include newer versions

### CODE REVIEW

File: .php-version [hunk 1]
```diff
@@ -0,0 +1 @@
+8.2
```
Comment (loevgaard) [inline .php-version]: This affects the Symfony binary, so if we in the future use that in the build it could have unintended side effects.

File: infection.json.dist [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "source": {
+        "directories": [
+            "src"
+        ]
+    },
+    "logs": {
+        "text": "php://stderr",
+        "github": true,
+        "stryker": {
+            "badge": "1.12.x"
```
Comment (loevgaard) [inline infection.json.dist]: This should be changed to `0.4.x`

File: tests/Application/config/packages/_sylius.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ imports:
     - { resource: "@SyliusCoreBundle/Resources/config/app/config.yml" }
 
     - { resource: "@SyliusAdminBundle/Resources/config/app/config.yml" }
-    - { resource: "@SyliusAdminApiBundle/Resources/config/app/config.yml" }
+#    - { resource: "@SyliusAdminApiBundle/Resources/config/app/config.yml" }
```
Comment (loevgaard) [inline tests/Application/config/packages/_sylius.yaml]: Just remove this line

File: tests/Application/.env.test [hunk 4]
```diff
@@ -1,3 +1,3 @@
 APP_SECRET='ch4mb3r0f5ecr3ts'
 
-KERNEL_CLASS='Tests\Setono\SyliusKlaviyoPlugin\Application\Kernel'
+KERNEL_CLASS='Tests\Acme\SyliusExamplePlugin\Application\Kernel'
```
Comment (loevgaard) [inline tests/Application/.env.test]: Revert these changes. Not that it matters I guess, just for consistency :)

File: README.md [hunk 5]
```diff
@@ -95,6 +95,10 @@ framework:
 After setup you want to associate channels with lists in Klaviyo. Go to `/admin/klaviyo/member-lists/`
 and synchronize the lists. After that edit the lists and enable channels on the lists.
 
+//todo:
```
Comment (loevgaard) [inline README.md]: Let's rewrite the README later. Just revert these changes in this PR

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+version: "3"
```
Comment (loevgaard) [inline docker-compose.yaml]: Shouldn't this be in the `tests/Application` directory to fully work with the Symfony binary? Also I have never added `docker-compose.yaml` files to my plugins, but I am not against it per se - there should just be a section in the `README.md` about it then.

**Comment by JoppeDC** [comment]
@loevgaard Thanks a lot for the extensive feedback. I did the adjustments as requested :)

**Comment by loevgaard** [comment]
Great work, @JoppeDC. Thank you 🎉
